pkg/kubelet/kubelet.go

```diff
@@ -1909,6 +1932,8 @@ func (kl *Kubelet) syncLoopIteration(configCh <-chan kubetypes.PodUpdate, handle
 				klog.Errorf("Failed cleaning pods: %v", err)
 			}
 		}
+	case <-watchdogCh:
```

Comment:
Are we 100% sure that this is the correct place to do this? I think i'd prefer if you could lay out some test cases and create e2e tests to make sure it works as expected. Are we sure that we don't end up killing a healthy kubelet before it properly starts, or when the apiserver is down eg.?

Would also be nice with a writeup about when we intend to stop reporting `SdNotify` (and when we do want to report it), to make it easier to debug in case it behaves in a non-understandable manner. Since the env-var is in micro secs I am a bit concerned that small values may results in weird behavior, but that is just a thought.

Reply:
- I will check if I find a better place for this code, and add tests as soon as I'm more familiar with the kubelet / test code
- We should stop notifying when kubelet is not healthy anymore. Will see if I find something useful to determine this. If you have ideas, please let me know
- the env var is in micro seconds, but the value which actually is set in the service config is in seconds, so I'm less concerned about this (even though the smallest value of 1 sec still is not that much...). See https://www.freedesktop.org/software/systemd/man/systemd.service.html#WatchdogSec=